src/Root.tsx

```diff
@@ -215,17 +215,24 @@ export default () => {
   const appLanguage = useAppSelector(({APP}) => APP.defaultLanguage);
   const pinLockActive = useAppSelector(({APP}) => APP.pinLockActive);
   const showBlur = useAppSelector(({APP}) => APP.showBlur);
+  const failedAppInit = useAppSelector(({APP}) => APP.failedAppInit);
   const biometricLockActive = useAppSelector(
     ({APP}) => APP.biometricLockActive,
   );
   const lockAuthorizedUntil = useAppSelector(
     ({APP}) => APP.lockAuthorizedUntil,
   );
+  const [initFailed, setInitFailed] = useState(false as boolean);
 
   // MAIN APP INIT
   useEffect(() => {
-    dispatch(AppEffects.startAppInit());
-  }, [dispatch]);
+    if (!failedAppInit) {
+      dispatch(AppEffects.startAppInit());
+    } else {
+      setInitFailed(true);
+      navigationRef.navigate(RootStacks.DEBUG, {name: 'Failed app init'});
```

Comment:
Depending where the error occurs, navRef may not be ready in the Root component yet, try navigating from the catch block in the effect

Reply:
Another quick solution would be to move the `await sleep(2000);` to before the `dispatch(AppActions.failedAppInit());` to give `navigationRef` enough time to get ready (this fix worked for me no matter where the error was thrown)

Reply:
Yes, I agree. navRef may not be ready if error occurs early. But I prefer to add sleep before call failedAppInit since I don't like to add a navigation action into the effect.

Thank you for your suggestions!